src/CovidSim.cpp

```diff
@@ -2567,25 +2568,25 @@ void InitModel(int run) // passing run number so we can save run number in the i
 	{
 		for (i = tn; i < P.NC; i+=P.NumThreads)
 		{
-			if ((Cells[i].tot_treat != 0) || (Cells[i].tot_vacc != 0) || (Cells[i].S != Cells[i].n) || (Cells[i].D > 0) || (Cells[i].R > 0))
+			if ((Cells.at(i).tot_treat != 0) || (Cells.at(i).tot_vacc != 0) || (Cells.at(i).S != Cells.at(i).n) || (Cells.at(i).D > 0) || (Cells.at(i).R > 0))
```

Comment:
`std::vector::at()` will throw an exception on a bad access. Not a bad thing per se - however, the code isn't exception aware and I think there will be problems with exceptions crossing thread boundaries.

There are two solutions to this:

The easy one: Just use `[]` access everywhere.  (Why do you use it in some places and not others?).

The harder one, put `try`/`catch` blocks everywhere that is appropriate.